x-pack/plugin/ent-search/qa/rest/src/yamlRestTest/resources/rest-api-spec/test/entsearch/55_search_application_search.yml

```diff
@@ -272,3 +297,24 @@ teardown:
           params:
             field_value: puggles
 
+---
+"Search application search fails on invalid rendered JSON":
+  - skip:
+      features: headers
+
+  - do:
+      catch: "bad_request"
```

Comment:
Can we check the message here?

Reply:
Not sure how, any examples we can pull from?

Reply:
You can use regexes for checking the message, like [here](https://github.com/elastic/elasticsearch/blob/2abfa6c06ce96b054a49e3e36c4016ed753ba048/rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/synonyms/100_synonyms_no_system_index.yml#L17) or [here](https://github.com/elastic/elasticsearch/blob/037261356e4ecbd1b7b91611ad9c9171dafc9b89/rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/mget/13_missing_metadata.yml#L12).

AFAIU we can't perform both a status code check and a message check - I'd maintain the status code check here as that is the main purpose of this change.

Reply:
I'm OK with the status code, messages can be a bit more clear, as you can guarantee the specific error was thrown.